overlord/boot/firstboot.go

```diff
@@ -122,14 +123,20 @@ func populateStateFromInstalled() error {
 	return ovld.Stop()
 }
 
+// replaced in the tests
+var firstbootEnableFirstEther = firstboot.EnableFirstEther
+
 // FirstBoot will do some initial boot setup and then sync the
 // state
 func FirstBoot() error {
 	if firstboot.HasRun() {
 		return ErrNotFirstBoot
 	}
-	if err := firstboot.EnableFirstEther(); err != nil {
-		logger.Noticef("Failed to bring up ethernet: %s", err)
+
+	if !release.OnClassic {
+		if err := firstbootEnableFirstEther(); err != nil {
+			logger.Noticef("Failed to bring up ethernet: %s", err)
+		}
 	}
 
 	// snappy will be in a very unhappy state if this happens,
```

Comment:
Do you want to skip the whole `FirstBoot()` code or do you really want to run `populateStateFromInstalled()`?